docs/ARCHITECTURE.md

```diff
@@ -0,0 +1,91 @@
+# MynahUI Architecture
+
+## How the Consumer and MynahUI works together?
```

Comment:
How do the Consumer and MynahUI work together?**
(small nit for better wording, rest of the comments as well)